src/config/mongoose_config_validator_toml.erl

```diff
@@ -397,16 +397,1133 @@ validate([<<"max_retry_delay">>, <<"s2s">>|Path],
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{node_type, V}]) ->
+    validate_non_empty_atom(V);
+validate([<<"notification_type">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{notification_type, V}]) ->
+    validate_non_empty_atom(V);
+validate([<<"notify_config">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{notify_config, V}]) ->
+    validate_boolean(V);
+validate([<<"notify_delete">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{notify_delete, V}]) ->
+    validate_boolean(V);
+validate([<<"notify_retract">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{notify_retract, V}]) ->
+    validate_boolean(V);
+validate([<<"persist_items">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{persist_items, V}]) ->
+    validate_boolean(V);
+validate([<<"presence_based_delivery">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{presence_based_delivery, V}]) ->
+    validate_boolean(V);
+validate([<<"publish_model">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{publish_model, V}]) ->
+    validate_non_empty_atom(V);
+validate([<<"purge_offline">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{purge_offline, V}]) ->
+    validate_boolean(V);
+validate([item, <<"roster_groups_allowed">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [V]) ->
+    validate_non_empty_binary(V);
+validate([<<"send_last_published_item">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{send_last_published_item, V}]) ->
+    validate_non_empty_atom(V);
+validate([<<"subscribe">>, <<"default_node_config">>,
+          <<"mod_pubsub">>, <<"modules">>|_],
+         [{subscribe, V}]) ->
+    validate_boolean(V);
+validate([<<"host">>, <<"mod_pubsub">>, <<"modules">>|_],
+         [{host, V}]) ->
+    validate_domain_template(V);
+validate([<<"ignore_pep_from_offline">>, <<"mod_pubsub">>, <<"modules">>|_],
+         [{ignore_pep_from_offline, V}]) ->
+    validate_boolean(V);
+validate([<<"iqdisc">>, <<"mod_pubsub">>, <<"modules">>|_],
+         [{iqdisc, V}]) ->
+    validate_iqdisc(V);
+validate([<<"item_publisher">>, <<"mod_pubsub">>, <<"modules">>|_],
+         [{item_publisher, V}]) ->
+    validate_boolean(V);
+validate([<<"last_item_cache">>, <<"mod_pubsub">>, <<"modules">>|_],
+         [{last_item_cache, V}]) ->
+    validate_enum(V, [mnesia,rdbms,false]);
+validate([<<"max_items_node">>, <<"mod_pubsub">>, <<"modules">>|_],
+         [{max_items_node, V}]) ->
+    validate_non_negative_integer(V);
+validate([<<"max_subscriptions_node">>, <<"mod_pubsub">>, <<"modules">>|_],
+         [{max_subscriptions_node, V}]) ->
+    validate_non_negative_integer(V);
```

Comment:
According to `mod_pubsub.md` we don't cover default value here.
```
default: `undefined` - no limitation
```

Reply:
The same will apply to `mod_websockets.ping_rate` field and its default `none` value.

Reply:
oh, nice catch, there is no validator for ping_rate.

Though it would really good to refactor undefined to infinity someday.

I think there is really easy way to set option to undefined in TOML - don't specify it :trollface: